test/k8sT/Nightly.go

```diff
@@ -269,62 +279,164 @@ var _ = Describe("NightlyExamples", func() {
-		defer fp.Close()
-		Expect(err).To(BeNil())
+		waitEndpointReady := func() {
+			ciliumPods, err := kubectl.GetCiliumPods(helpers.KubeSystemNamespace)
+			Expect(err).To(BeNil(), "cannot retrieve cilium pods")
+			for _, pod := range ciliumPods {
+				ExpectWithOffset(1, kubectl.CiliumEndpointWait(pod)).To(BeTrue(),
+					"Pods are not ready in pod %v", pod)
+			}
+		}
 
-		fmt.Fprint(fp, result.String())
+		AfterEach(func() {
+			kubectl.Exec(fmt.Sprintf(
+				"%s -n %s delete ds cilium",
+				helpers.KubectlCmd, helpers.KubeSystemNamespace)).ExpectSuccess(
+				"Cilium DS cannot be deleted")
+		})
 
-		kubectl.Apply(helpers.GetFilePath(newCiliumDSName)).ExpectSuccess()
-		defer kubectl.Delete(helpers.GetFilePath(newCiliumDSName))
-		status, err := kubectl.WaitforPods(helpers.KubeSystemNamespace, "-l k8s-app=cilium", 300)
-		Expect(status).Should(BeTrue())
-		Expect(err).Should(BeNil())
+		BeforeEach(func() {
+			kubectl.Exec("sudo docker rmi cilium/cilium")
+			InstallExampleCilium()
+		})
 
-		kubectl.Apply(demoPath).ExpectSuccess()
-		_, err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
-		Expect(err).Should(BeNil())
+		It("Check Kubernetes Example is working correctly", func() {
+			kubectl.Apply(demoPath).ExpectSuccess()
+			_, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+			Expect(err).Should(BeNil())
 
-		_, err = kubectl.CiliumPolicyAction(helpers.KubeSystemNamespace, l3Policy, helpers.KubectlApply, 300)
-		Expect(err).Should(BeNil())
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, l3Policy, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil())
 
-		appPods := getAppPods()
+			appPods := getAppPods()
 
-		clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, appService)
-		Expect(err).Should(BeNil())
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, appService)
+			Expect(err).Should(BeNil())
 
-		_, err = kubectl.ExecPodCmd(
-			helpers.DefaultNamespace, appPods[helpers.App2],
-			helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-		Expect(err).Should(BeNil())
+			_, err = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			Expect(err).Should(BeNil())
+
+			_, err = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			Expect(err).Should(HaveOccurred())
+		})
+
+		It("Updating Cilium stable to master", func() {
+			By("Creating some endpoints and L7 policy")
+			kubectl.Apply(demoPath).ExpectSuccess()
+			_, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+			Expect(err).Should(BeNil())
+
+			kubectl.Exec(fmt.Sprintf("%s scale deployment %s --replicas=10",
+				helpers.KubectlCmd, helpers.App1)).ExpectSuccess("Cilium DS cannot be deleted")
```

Comment:
If the command is "scale deployment", why does the message in `ExpectSuccess` say "Cilium DS cannot be deleted"?